src/Idris/REPL.idr

```diff
@@ -715,14 +749,23 @@ process (Exec ctm)
     = execExp ctm
 process Help
     = pure RequestedHelp
-process (ProofSearch n_in)
-    = do defs <- get Ctxt
-         [(n, i, ty)] <- lookupTyName n_in (gamma defs)
-              | [] => throw (UndefinedName replFC n_in)
-              | ns => throw (AmbiguousName replFC (map fst ns))
-         tm <- search replFC top False 1000 n ty []
-         itm <- resugar [] !(normaliseHoles defs [] tm)
-         pure $ ProofFound itm
+process (TypeSearch searchTerm@(PPi _ _ _ _ _ _))
+    = do defs <- branch
+         let ctxt = gamma defs
+         rawTy <- desugar AnyExpr [] searchTerm
+         let bound = piBindNames replFC [] rawTy
+         (ty, _) <- elabTerm 0 InType [] (MkNested []) [] bound Nothing
+         ty' <- toResolvedNames ty
+         filteredDefs <- do names   <- allNames ctxt
+                            defs    <- pure $ catMaybes !(traverse (flip lookupCtxtExact ctxt) names)
```

Comment:
Small syntactic suggestion, but I think this works:

```suggestion
                            defs    <- catMaybes <$> (traverse (flip lookupCtxtExact ctxt) names)
```

Reply:
Thats definitely better; thanks!